packages/plugin-ext/src/hosted/browser/hosted-plugin.ts

```diff
@@ -161,7 +163,7 @@ export class HostedPluginSupport {
                 workspaceStates: metadata['5'],
                 roots: metadata['6']
             };
-            this.loadPlugins(pluginsInitData, this.container);
+            this.loadingQueue.push(() => this.loadPlugins(pluginsInitData, this.container));
```

Comment:
I wonder why `loadPlugins` is public? if it can be called not from `initPlugins` then it will go around the queue.

I also wonder whether we need a queue in this case. I think maybe we have a map of loaded plugins. Where key is a plugin id and a value is a metadata. So in `loadPlugins` we immediately add new plugins into this map if they are not there yet, if they are we filter them out. If after that plugins to load is empty we do nothing, it there is something we update `plugins` of init data and proceed. So loading can happen in parallel, but not for the same plugin, also if it fails it won't try to load again.

@benoitf sounds good?

Reply:
@akosyakov Yes, it's good for me.